lib/p2p/packets/types/DisconnectingPacket.ts

```diff
@@ -15,6 +18,46 @@ class DisconnectingPacket extends Packet<DisconnectingPacketBody> {
   public get direction() {
     return PacketDirection.Unilateral;
   }
+
+  public static deserialize(binary: Uint8Array): DisconnectingPacket | undefined {
+    const msg = pb.DisconnectingPacket.deserializeBinary(binary).toObject();
+    return DisconnectingPacket.validate(msg) ? DisconnectingPacket.convert(msg) : undefined;
+  }
+
+  private static validate(msg: pb.DisconnectingPacket.AsObject): boolean {
+    return !!(msg.header
```

Comment:
For these, I don't think we need the `!!` when we're using the `&&` operators already. It doesn't hurt to have it, but it seems a bit cleaner without.

Reply:
It's needed because otherwise the return type won't be a boolean.

Reply:
Yes actually you are right, my mistake.